src/core/plugin/luapi_application.h

```diff
@@ -327,6 +384,263 @@ static int applib_layerAction(lua_State* L) {
     return 1;
 }
 
+static int addStrokeHelper(lua_State* L, Stroke* stroke, Layer* layer) {
+    //    Plugin* plugin = Plugin::getPluginFromLua(L);
+    //    Control* ctrl = plugin->getControl();
+    //    PageRef const& page = ctrl->getCurrentPage();
+    //    Layer* layer = page->getSelectedLayer();
+    //    Stroke* stroke = new Stroke();
+
+    // discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TTABLE);
+
+    // Make sure we have enough points to form a stroke
+    if (stroke->getPointCount() >= 2) {
+        // Get the rest of the attributes. Default to something if none is provided
+        // TODO: (willnilges) Test this. Is this robust? Should we be doing this one field at a time?
+        lua_getfield(L, 1, "width");
+        lua_getfield(L, 1, "color");
+        lua_getfield(L, 1, "fill");
+        lua_getfield(L, 1, "tool");
+        lua_getfield(L, 1, "lineStyle");
+        double width = luaL_optnumber(L, -5, 1.5);
```

Comment:
Ah, thanks! Didn't realize you could do that. 1.5 was chosen out of ignorance :sweat_smile:

Will change :+1:

Reply:
Now the color and size, if no others are passed, are based off of the currently selected options. Thought that might be a nice touch.

I was thinking that maybe, by default, everything should be based off of current options.

Reply:
The helper will now check what the current pen thickness and color are, and apply those options if there were none provided.

Going further, I think it might be best to just default entirely to the current pen options.